library/ssl_cli.c

```diff
@@ -1232,6 +1244,7 @@ static int ssl_parse_supported_point_formats_ext( mbedtls_ssl_context *ssl,
     }
 
     MBEDTLS_SSL_DEBUG_MSG( 1, ( "no point format in common" ) );
+        mbedtls_ssl_send_fatal_handshake_failure( ssl );
```

Comment:
I agree most of the time, but there is one error that we might want to handle in an ideal world: `MBEDTLS_ERR_SSL_WANT_WRITE` (which usually means that we actually didn't send the alert out, in which can we might want to retry).

However, that's a long-standing issue with all alerts we send, for which I don't see an easy solution (we could remember the original error somewhere in the SSL context, but that means ABI change), which I think is quite orthogonal to this PR and can be addressed separately.